src/main/scala/decider/Z3ProverStdIO.scala

```diff
@@ -6,294 +6,43 @@
 
 package viper.silicon.decider
 
-import java.io._
 import java.nio.file.{Path, Paths}
-import java.util.concurrent.TimeUnit
 
-import com.typesafe.scalalogging.LazyLogging
-import viper.silicon.common.config.Version
-import viper.silicon.interfaces.decider.{Prover, Sat, Unknown, Unsat}
-import viper.silicon.reporting.{ExternalToolError, Z3InteractionFailed}
 import viper.silicon.state.IdentifierFactory
-import viper.silicon.state.terms._
 import viper.silicon.verifier.Verifier
-import viper.silicon.{Config, Map, toMap}
-import viper.silver.reporter.{ConfigurationConfirmation, InternalWarningMessage, Reporter}
+import viper.silver.verifier.{DefaultDependency => SilDefaultDependency}
+import viper.silver.reporter.Reporter
+import viper.silicon.common.config.Version
+
+
+object Z3ProverStdIO {
+  val name = "Z3"
+  val minVersion = Version("4.5.0")
+  val maxVersion = None // Some(Version("4.5.0")) /* X.Y.Z if that is the *last supported* version */
```

Comment:
I thought there were actual problems with higher versions @mschwerhoff ?

Reply:
I cannot offer any input on this since I have not tested it. The `minVersion` and `maxVersion` are just moved from Silicon.scala and the values are kept the same.

Reply:
@Aurel300  `maxVersion` is currently disabled because no Z3 version really brakes Silicon - whereas Z3 prior to 4.5.0 isn't supported anymore due to configuration option changes. Currently, newer versions make a specify subset of our tests fail (old VerCors examples we suspect of matching loops), but ignoring those, newer versions are supported.